Caching

nguyenfilip opened the pull request:
Documentation for 2LC and JCache usage in Candlepin

kahowell commented on docs/candlepin/caching.md:
```diff
@@ -0,0 +1,75 @@
+We use two kinds of caches
+  
+ * JCache (JSR-107) - this caching standard is useful for caching key-value data
+ * Hibernate Second Level Cache (2LC) - 
```
Trailing `-`, or maybe you didn't finish your thought?

kahowell commented on docs/candlepin/caching.md:
````diff
@@ -0,0 +1,75 @@
+```
+{:.numbered}
+
+Note that Cache interface is from JSR-107.
````
You could expand upon how we're deliberately using the JSR-107 interfaces, and *not* using EHCache interfaces, so that we could in theory swap out EHCache for another implementation in the future if we wanted.

kahowell commented on docs/candlepin/caching.md:
```diff
@@ -0,0 +1,75 @@
+Note that Cache interface is from JSR-107.
+
+### Using 2LC - Hibernate 2nd Level Cache
+To cache entities using 2LC, it is necessary to use appropriate annotations: 
```
Might be useful to link to https://docs.jboss.org/hibernate/orm/3.3/reference/en-US/html/performance.html (or even better: a version corresponding to our Hibernate version).

kahowell commented:
Overall, looks great!

This is my preference, but I like to surround all class names, file names, and file paths in back-ticks (`). I notice that we do this some places, but not consistently one way or the other. I think it increases readability because it visually highlights code-related names.

Unrelated to this PR, I am thinking we should revisit this page later on, once we're using caching for specific data, and document or at least link documentation from this page.

nguyenfilip commented:
@kahowell Kevin, thank you for great feedback. I agree with all the points you raised. I also added more Hibernate links to the beginning of the document. For the convenience, I added the review feedback changes as a second commit. Tell me if you are fine with them and I will squash the commits.

kahowell commented:
@nguyenfilip, changes look great! Let's squash and merge.

nguyenfilip commented:
@kahowell done!
